server.go

```diff
@@ -980,7 +980,7 @@ func (s *Server) ShardGroups(database string) ([]*ShardGroup, error) {
 
 // CreateShardGroupIfNotExists creates the shard group for a retention policy for the interval a timestamp falls into.
 func (s *Server) CreateShardGroupIfNotExists(database, policy string, timestamp time.Time) error {
-	c := &createShardGroupIfNotExistsCommand{Database: database, Policy: policy, Timestamp: timestamp}
+	c := &createShardGroupIfNotExistsCommand{Database: database, Policy: policy, Timestamp: timestamp, URL: s.URL()}
```

Comment:
This was to avoid a deadlock in the server when processing this command.  The issue is that s.URL() locks the mutex that is already locked here: https://github.com/influxdb/influxdb/blob/master/server.go#L3190